groupease-server/src/main/java/io/github/groupease/channelmember/DefaultChannelMemberService.java

```diff
@@ -126,27 +126,28 @@ public Member update(
         );
 
         /*
-         * TODO: Get member & confirm access. See delete method below for example.
-         * Additional check:
          * Ensure if "isOwner" is being changed, that current user is a channel admin.
+         * Also ensure non-owners can only edit their own profile.
          */
 
         Member editor = getForCurrentUser(toUpdate.getChannel().getId());
-        Member existingMember = getByMemberId(toUpdate.getId());
-
-        if(editor.isOwner() || editor.getId().equals(toUpdate.getId())){   //Is the current user an owner? or editing themself?
-            if(toUpdate.isOwner() == existingMember.isOwner()){ //Changing isOwner?
-                if(editor.isOwner()){
```

Comment:
This line (and the one above) added in Issue #70 caused a bug that prohibited any non-user from editing their profile.  It always would throw `ChannelMemberNotOwnerException` on line 142 below.